### PR TITLE
[SPARK-47330][SQL][TESTS] XML: Added XmlExpressionsSuite

### PR DESCRIPTION
**Pull request by yhosny**
### What changes were proposed in this pull request?

Replicated JsonExpressionsSuite to XmlExpressionsSuite for additional XML unit test coverage.

### Why are the changes needed?

Improve XML Expressions unit testing coverage.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Uni tests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Review by MaxGekk — CHANGES_REQUESTED** [review]
@yhosny Could you fix coding style, please:
```
[error] /home/runner/work/spark/spark/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/XmlExpressionsSuite.scala:22:0: There should at least one a single empty line separating groups java and 3rdParty.
[error] /home/runner/work/spark/spark/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/XmlExpressionsSuite.scala:23:0: There should at least one a single empty line separating groups 3rdParty and spark.
```

**Review by shujingyang-db — CHANGES_REQUESTED** [review]
blocking this OSS PR...Will discuss the test cases offline.

**Comment by HyukjinKwon** [comment]
@sandip-db does it LG to you?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/XmlExpressionsSuite.scala [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+                     |    <f>
+                     |        <a>1</a>
+                     |    </f>
+                     |</ROW>""".stripMargin
+    checkEvaluation(
+      StructsToXml(Map.empty, Literal.create(input, inputSchema), UTC_OPT),
+      output)
+  }
+
+  test("from/to xml - interval support") {
```
Comment (sandip-db) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/XmlExpressionsSuite.scala:335]: remove this test as CalendarIntervalType is not supported yet.

**Review by sandip-db — CHANGES_REQUESTED** [review]

**Review by shujingyang-db — APPROVED** [review]

**Review by sandip-db — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**